code/modules/mob/living/living.dm

```diff
@@ -443,6 +443,12 @@
 	if (s_active && !(s_active.ClickAccessible(src, depth=STORAGE_VIEW_DEPTH) || s_active.Adjacent(src)))
 		s_active.close(src)
 
+	for(var/mob/M in oview(src))
+		if(M.client && (src in cone(M, OPPOSITE_DIR(M.dir), view(10, M))))
+			M.update_vision_cone()
```

Comment:
Could do it on the player list instead of the mob list, would be significantly smaller